src/main/java/com/diffmin/App.java

```diff
@@ -209,6 +211,13 @@ private static void applyInsertion(ImmutableTriple<Integer, CtElement, CtElement
                 ((CtExecutable<?>) inWhichElement)
                         .addParameterAt(where, (CtParameter<?>) toBeInserted);
                 break;
+            case THROWN:
+                ((CtExecutable<Throwable>) inWhichElement)
+                        .addThrownType((CtTypeReference<Throwable>) toBeInserted);
+                //                ((CtExecutable<?>)
+                // inWhichElement).setThrownTypes(((CtExecutable<?>)
+                // toBeInserted.getParent()).getThrownTypes());
```

Comment:
No idea what causes this, you'll have to do some debugging here :). But I think that this is the most appropriate approach to take here as we otherwise lose control of the order of the elements.

Reply:
I figured this out. I will try to explain thoroughly what is happening across the two patches.

### 1st patch

```
inWhichElement = public void func() throws SecurityException {}
inWhichElement.getThrownTypes() = [SecurityException] // id=1 (memory location)
toBeInserted = InstantiationError
toBeInserted.getParent().getThrownTypes = [InstantiationError, IllegalStateException, SecurityException] // id=2 (memory location)
```

`inWhichElement.setThrownTypes(toBeInserted.getParent().getThrownTypes)` works well for the first patch and `inWhichElement.getThrownTypes()`'s id also becomes equal to `2`.

### 2nd patch

```
inWhichElement = public void func() throws InstantiationError, IllegalStateException, SecurityException { }
inWhichElement.getThrownTypes() = [InstantiationError, IllegalStateException, SecurityException] // id=2 (memory location)
toBeInserted = IllegalStateException
toBeInserted.getParent().getThrownTypes = [InstantiationError, IllegalStateException, SecurityException] // id=2 (memory location)
```

So when the method `setThrownTypes` is called during the application of the 2nd patch, [`this.thrownTypes.clear()`](https://github.com/INRIA/spoon/blob/master/src/main/java/spoon/support/reflect/declaration/CtExecutableImpl.java#L149) is also called. And as `this.thrownTypes` and `thrownTypes` (parameter of `setThrownTypes`) refer to the **same memory location (id=2)**, the `thrownTypes` from the method is erased! That is why were getting no method signatures after the second patch.

I hope this explanation is clear but if it is not I can propose an alternate shorter example.
```java
method.setThrownTypes(method.getThrownTypes());
``` 
Since `this.thrownTypes` and `thrownTypes` point to the same location, both will get cleared leaving us a method without any signatures even if there were before.


### Solution

Before setting the thrownTypes, we can check if the above sets point to the same location or not.